cirq-web/cirq_ts/src/circuit/components/meshes.ts

```diff
@@ -0,0 +1,227 @@
+   * @param startCoord The starting coordinate of the line
+   * @param endCoord The ending coordinate of the line
+   * @returns a QubitLine object that can be added to a three.js scene.
+   */
+  constructor(startCoord: Vector3, endCoord: Vector3) {
+    const COLOR = 'gray';
+
+    const material = new LineBasicMaterial({color: COLOR});
+    const points = [startCoord, endCoord];
+    const geometry = new BufferGeometry().setFromPoints(points);
+
+    super(geometry, material);
+    return this;
+  }
+}
+
+/**
+ * A wrapper for a three.js Sphere which represents a control
+ * in a circuit.
+ */
+export class Control3DSymbol extends Mesh {
+  /**
+   * Class constructor.
+   * @returns a Control3DSymbol object that can be added to a three.js scene.
+   */
+  constructor() {
+    const COLOR = 'black';
+
+    const material = new MeshBasicMaterial({color: COLOR});
+    const geometry = new SphereGeometry(0.1, 32, 32);
+
+    super(geometry, material);
+    return this;
+  }
+}
+
+/**
+ * A wrapper for a three.js Cylinder which represents an X operation
+ * in a circuit.
+ */
+export class X3DSymbol extends Group {
+  /**
+   * Class constructor.
+   * @param color The color of the symbol
+   * @returns an X3DSymbol object that can be added to a three.js scene
+   */
+  constructor(color: string) {
+    super();
+    const material = new MeshBasicMaterial({color: color, side: DoubleSide});
+    const geometry = new CylinderGeometry(
+      0.3,
+      0.3,
+      0.1,
+      32,
+      1,
+      true,
+      0,
+      2 * Math.PI
+    );
+    const hollowCylinder = new Mesh(geometry, material);
+    this.add(hollowCylinder);
+
+    // Creates the "X" in the middle of the holow cylinder
+    const rotationAngle = Math.PI / 2;
+
+    const xLineMaterial = new MeshBasicMaterial({color: color});
+    const xLineGeometry = new CylinderGeometry(0.01, 0.01, 0.6);
+    const xLine = new Mesh(xLineGeometry, xLineMaterial);
+    xLine.rotation.x = rotationAngle;
+
+    const zLineMaterial = new MeshBasicMaterial({color: color});
+    const zLineGeometry = new CylinderGeometry(0.01, 0.01, 0.6);
+    const zLine = new Mesh(zLineGeometry, zLineMaterial);
+    zLine.rotation.z = rotationAngle;
+
+    this.add(xLine);
+    this.add(zLine);
+    return this;
+  }
+}
+
+/**
+ * A wrapper for a three.js Box which represents arbitrary single qubit
+ * operations in a circuit
+ */
+export class BoxGate3DSymbol extends Mesh {
+  /**
+   * Class constructor.
+   * @param label The label that will go on the three.js box
+   * @param color The color of the box
+   * @returns a BoxGate3DSymbol object that can be added to a three.js scene
+   */
+  constructor(label: string, color: string) {
+    const canvas = document.createElement('canvas')!;
+    const context = canvas.getContext('2d')!;
+    canvas.width = canvas.height = 128;
+
+    context.fillStyle = color;
+    context.fillRect(0, 0, canvas.width, canvas.height);
+
+    // TODO: Make this more robust
```

Comment:
Create an issue and reference it in the TODO

Reply:
I fixed this one quickly